### DISCUSSION TITLE
Weird feedback loop, and record both mic and mac audio simultaneously?

### DISCUSSION THREAD

**Discussion by fknutas** [ExistentialAudio/BlackHole#642] [General]
Hey

I'm new to this but i've followed the installer multiple times trying to figure this one out.

I'm a podcaster that wants to 
- record internal audio from my mac
- record audio from a mic that is connected to a presonus sound card
- record both these in the same audio track in garageband
- monitor the recording in garageband using my external headphones connected to the aux in my mac

I've set my gear and BlackHole up in this fashion:

Multi output device: external headphones and blackhole 2ch
Aggregate device: Presonus audiobox and blackhole 2ch

When i try to record the mac internal audio on one channel, i get this insane feedback loop, and when i try recording the mic i get nothing. When i try the other channel, the mic is fine but no mac sounds gets recorded.
I've tried lumping them all into both multi output and aggregate but nothing helps.

Do i need to get the version with more channels or does this have to do with something else?

Thank you for an amazing software which when it works i bet works perfectly. Genius move making this when Apple fell short on us audio recorders.

**Comment by devinroth** [comment]
I suspect GarageBand is your limiting factor in this setup. My understanding is GarageBand only works with 2 channels which makes complex routing impossible.

I have a similar setup with Logic Pro. I use one aggregate device for all the audio devices and setup aux tracks to route audio to wherever I need it to go. And then just record whatever I want.

Devin
